caliban_toolbox/aws_functions.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2016-2020 David Van Valen at California Institute of Technology
+# (Caltech), with support from the Paul Allen Family Foundation, Google,
+# & National Institutes of Health (NIH) under Grant U24CA224309-01.
+# All rights reserved.
+#
+# Licensed under a modified Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.github.com/vanvalenlab/caliban-toolbox/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import sys
+import boto3
+import os
+import threading
+import re
+
+from urllib.parse import urlencode
```

Comment:
I expect this to have some issues with py2 vs py3 compatibility. You can check the redis-consumer to see how we handled it before.

Reply:
So currently the toolbox is only being built against 3.6. 

Do you think we should expand? I understand the rationale for deepcell-tf, which is much more widely used. Not so sure it makes sense here, especially since they just released the last ever 2.X build.

Reply:
Good point. I forgot which project I was in.  Probably no need to build for py2, but we should probably build it against a few python3 versions (3.5, 3.6, 3.7?)